src/main/java/appeng/fluids/helper/DualityFluidInterface.java

```diff
@@ -653,10 +578,37 @@ public IAEFluidStack extractItems( final IAEFluidStack request, final Actionable
 		}
 	}
 
-	@Override
 	public void saveChanges()
 	{
 		this.iHost.getTileEntity().markDirty();
 	}
 
+	@Override
+	public IConfigManager getConfigManager()
+	{
+		return this.cm;
+	}
+
+	@Override
+	public IItemHandler getInventoryByName( String name )
+	{
+		return null;
+	}
+
+	@Override
+	public int getInstalledUpgrades( Upgrades u )
+	{
+		return 0;
+	}
+
+	@Override
+	public TileEntity getTile()
+	{
+		return (TileEntity) ( this.iHost instanceof TileEntity ? this.iHost : null );
+	}
+
+	@Override
+	public void updateSetting( IConfigManager manager, Enum settingName, Enum newValue )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)